### PR TITLE
feat(github-tags): Supplement with releases metadata

### LINKED ISSUE #7568: Merge github-releases metadata into github-tags
**What would you like Renovate to be able to do?**

Enrich github-tags datasource metadata with values from github-releases when both are present.

**Did you already have any implementation ideas?**

Any time github-tags is requested, the datasource should call its sibling github-releases too. Then for every tag that matches, add in any missing metadata. Perhaps possible to *replace* the tags object with releases one completely.. I'm not sure.

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**

If users switch from tags to releases then they may miss out on upgrading to any tags for which the repo forgot to create a release for.

**Is this a feature you'd be interested in implementing yourself?**

No

### PR DESCRIPTION
**Pull request by zharinov**
Closes #7568

### CODE REVIEW

File: lib/datasource/github-releases/index.ts [hunk 1]
```diff
@@ -35,13 +37,17 @@ export async function getReleases({
   if (cachedResult) {
     return cachedResult;
   }
-  const url = `https://api.github.com/repos/${repo}/releases?per_page=100`;
+  const sourceUrlBase = depHost ?? `https://github.com/`;
```
Comment (rarkins) [inline lib/datasource/github-releases/index.ts:40]: Can you separate this into a new PR? I'm surprised we didn't have a need for enterprise github releases before.

File: lib/datasource/github-tags/index.ts [hunk 2]
```diff
@@ -166,3 +157,34 @@ export async function getReleases({
   );
   return dependency;
 }
+
+function normalizeVersion(version: string): string {
+  return version?.replace(/^v/, '');
```
Comment (rarkins) [inline lib/datasource/github-tags/index.ts:162]: What about using the `semver` package's functions like this: `semver.valid(semver.coerce('v2')) // '2.0.0'` ?

File: lib/datasource/github-tags/index.ts [hunk 3]
```diff
@@ -166,3 +157,34 @@ export async function getReleases({
...
+}
+
+export async function getReleases(
+  config: GetReleasesConfig
+): Promise<ReleaseResult | null> {
+  const tagsResult = await getTags(config);
+
+  try {
+    // Fetch additional data from releases endpoint when possible
+    const releasesResult = await githubReleases.getReleases(config);
```
Comment (rarkins) [inline lib/datasource/github-tags/index.ts:172]: Is it possible to call via the datasource index? e.g. to benefit from indexing. Not sure if we'll end up with a circular dependency though.

File: lib/datasource/github-tags/index.ts [hunk 4]
```diff
@@ -166,3 +157,34 @@ export async function getReleases({
...
+    releasesResult?.releases?.forEach((release) => {
+      const key = normalizeVersion(release.version);
+      const value = { ...release };
+      delete value.version;
+      releaseByVersion[key] = value;
+    });
+
+    tagsResult?.releases?.forEach((tag) => {
+      const release = releaseByVersion[normalizeVersion(tag?.version)];
+      Object.assign(tag, release);
```
Comment (rarkins) [inline lib/datasource/github-tags/index.ts:183]: Do we definitely want to overwrite all fields, or cherry pick them selectively?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/datasource/github-tags/index.ts (hunk 2 above)
Comment (zharinov) [reply lib/datasource/github-tags/index.ts:162]: I didn't want to rely on particular version schemes in these datasources. `semver.coerce` should return wide range of normalized result strings seldom returning null, but `/^v/` stripping always returns string.

Comment (rarkins) [reply lib/datasource/github-tags/index.ts:162]: Actually I think GitHub releases raw result includes the exact tag, which would be a better way to match?

File: lib/datasource/github-tags/index.ts (hunk 3 above)
Comment (zharinov) [reply lib/datasource/github-tags/index.ts:172]: Maybe it's possible, but then we need to make decisions about `registryUrls` strategy. And yes, it looks like something circular to me too.

**Pull request merged**